### PR TITLE
Support null exception in ErrorMessage

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
Support construct ErrorMessage object with null exception.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dhrubo-os** [comment]
Could you please add more details in the PR description? Trying to understand when do we face this issue?

File: plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java [hunk 1]
```diff
@@ -44,14 +44,18 @@ public ErrorMessage(Throwable exception, int status) {
     }
 
     private String fetchType() {
-        return exception.getClass().getSimpleName();
+        return exception == null ? "NULL" : exception.getClass().getSimpleName();
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java:47]: If exception is null, should we return empty exception type ?

Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java:47]: what about `No Exception Provided` or `Unknown Exception`?

File: plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java [hunk 2]
```diff
@@ -44,14 +44,18 @@ public ErrorMessage(Throwable exception, int status) {
...
     }
 
     protected String fetchReason() {
         return status == RestStatus.BAD_REQUEST.getStatus() ? "Invalid Request" : "System Error";
     }
 
     protected String fetchDetails() {
+        if (exception == null) {
+            return "NULL";
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java:56]: `return "No Exception Details";` ?

Comment (jngz-es) [reply plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java:56]: Yeah, it's more readable.

File: plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java (hunk 1 above)
Comment (jngz-es) [reply plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java:47]: Sounds good, I will use `Unknown Exception`.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**